bdk_chain/src/chain_graph.rs

```diff
@@ -82,21 +63,42 @@ impl<P: ChainPosition> ChainGraph<P> {
         Some((position, full_tx))
     }
 
-    pub fn insert_output(&mut self, outpoint: OutPoint, txout: TxOut) -> bool {
-        self.graph.insert_txout(outpoint, txout)
+    /// Determines the changes required to insert a transaction into the inner [`ChainGraph`] and
+    /// [`SparseChain`] at the given `position`.
+    ///
+    /// If inserting it into the chain `position` will result in conflicts, the returned
+    /// [`ChangeSet`] should evict conflicting transactions.
+    pub fn insert_tx_changeset(
+        &self,
+        tx: Transaction,
+        pos: P,
+    ) -> Result<ChangeSet<P>, UpdateFailure<P>> {
```

Comment:
The error here is too general. `UpdateFailure` contains a `NotConnected` variant which cannot occur while inserting a tx. This needs to maintain its own error type.